whois-commons/src/main/java/net/ripe/db/whois/common/grs/AuthoritativeResourceWorker.java

```diff
@@ -0,0 +1,93 @@
+package net.ripe.db.whois.common.grs;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import net.ripe.commons.ip.Asn;
+import net.ripe.commons.ip.AsnRange;
+import net.ripe.commons.ip.Ipv4;
+import net.ripe.commons.ip.Ipv4Range;
+import net.ripe.commons.ip.Ipv6;
+import net.ripe.commons.ip.Ipv6Range;
+import net.ripe.commons.ip.SortedRangeSet;
+import org.slf4j.Logger;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.core.HttpHeaders;
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ExecutorService;
+
+public class AuthoritativeResourceWorker {
+
+    protected final Logger logger;
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    final SortedRangeSet<Asn, AsnRange> autNums = new SortedRangeSet<>();
+    final SortedRangeSet<Ipv4, Ipv4Range> ipv4Space = new SortedRangeSet<>();
+    final SortedRangeSet<Ipv6, Ipv6Range> ipv6Space = new SortedRangeSet<>();
+
+    private final String rsngBaseUrl;
+    private final Client client;
+    private final String apiKey;
+    private final ExecutorService executorService;
+
+    AuthoritativeResourceWorker(final Logger logger, final String rsngBaseUrl, final Client client, final ExecutorService executorService, final String apiKey) {
```

Comment:
Rename to ```RsngAuthoritativeResourceWorker``` for clarity? (It depends on the RSNG rir-stats response).

We could also rename ```RipeAuthoritativeResourceImportTask``` to ```RsngAuthoritativeResourceImportTask``` ?

Or, add JavaDoc comments to the top of the class explaining its puropose.

Reply:
I think reason t is named as RipeAuthoritativeResourceImportTask is because just just imports  RIPE source name resources

Reply:
OK! Thanks